components/FlexibleHeader/src/MDCFlexibleHeaderView.m

```diff
@@ -452,11 +452,7 @@ - (void)safeAreaInsetsDidChange {
       // status bar visibility then we ignore this safeAreaInsetsDidChange event.
       return;
     }
-    CGFloat safeAreaTop = MDCDeviceTopSafeAreaInset();
-    if (_currentSafeAreaInsetTop == safeAreaTop) {
```

Comment:
This check here is to prevent us from doing all the stuff we need to do when the safe area actually changes when it actually hasn't changed. When the flexible header is used in a UITableViewController (see the App Bar example that does that), this method gets called whenever you scroll the content. Removing it breaks the stretching behavior in cases like that, because we end up re-setting the insets of the content view.

Reply:
Fixing symptoms at this point but i added extra clauses to fix your use case.